contracts/utils/src/Utils.compact

```diff
@@ -15,42 +15,55 @@ module Utils {
    * @notice Midnight's burn address is represented as left<ZswapCoinPublicKey, ContractAddress>(default<ZswapCoinPublicKey>)
    * in Compact, so we've chosen to represent the zero address as this structure as well.
    *
+   * Disclosures:
+   *
+   * - Leaks the ZSwapCoinPublicKey or ContractAddress data of `keyOrAddress`
+   *
    * @param {Either<ZswapCoinPublicKey, ContractAddress>} keyOrAddress - The target value to check, either a ZswapCoinPublicKey or a ContractAddress.
    * @return {Boolean} - Returns true if `keyOrAddress` is zero.
    */
-  export pure circuit isKeyOrAddressZero(keyOrAddress: Either<ZswapCoinPublicKey, ContractAddress>): Boolean {
+  export circuit isKeyOrAddressZero(keyOrAddress: Either<ZswapCoinPublicKey, ContractAddress>): Boolean {
     return isContractAddress(keyOrAddress)
-    ? default<ContractAddress> == keyOrAddress.right
-    : default<ZswapCoinPublicKey> == keyOrAddress.left;
+    ? default<ContractAddress> == disclose(keyOrAddress.right)
+    : default<ZswapCoinPublicKey> == disclose(keyOrAddress.left);
   }
 
   /**
    * @description Returns whether `key` is the zero address.
    *
+   * Disclosures:
+   *
+   * - Leaks the ZSwapCoinPublicKey data of `key`
```

Comment:
```suggestion
   * - The ZSwapCoinPublicKey data of `key`.
```
Nit: Do we need to repeat "Leaks"? This communicates the same thing and is just as clear, no?

Reply:
Same applies to the rest of you agree